airflow/models/dagrun.py

```diff
@@ -609,6 +609,7 @@ def update_state(
                 self.data_interval_end,
                 self.dag_hash,
             )
+            session.flush()
```

Comment:
Overall lgtm, should this be down with the `merge` though so it happens regardless of the ultimate state?

Reply:
That was my initial thought but I don't know if there would be a regression when added there since it would apply to running `dagruns` as well. It use to have a `commit` down there, after `merge` but was removed here: https://github.com/apache/airflow/commit/73b9163a8f55ce3d5bf6aec0a558952c27dd1b55#diff-649fbbf224bab54417f03338c27d0fdb3c3336e53a522a13dfd9806c99f63137L377

cc: @ashb

Reply:
I just did move this down with `merge` and query count increased...If we are cool with it I will fix the tests